### PR TITLE
luan/ryo 38 render effectsmissiles

### PR DESCRIPTION
**Pull request by luan**
- feat: better direction and position handling
- perf: move sprite loading to events to avoid system spinning
- feat: missle rendering
- chore: improve ergonomics of moving bundle

### CODE REVIEW

File: ryot/src/bevy_ryot/drawing/mod.rs [hunk 1]
```diff
@@ -100,6 +103,27 @@ impl DrawingBundle {
...
+        Self::new(
+            CipLayer::Effect.into(),
+            tile_pos,
+            AppearanceDescriptor::effect(id),
+        )
+    }
+
+    pub fn missile(tile_pos: TilePosition, id: u32) -> Self {
+        Self::new(
+            CipLayer::Top.into(),
```
Comment (lgrossi) [inline ryot/src/bevy_ryot/drawing/mod.rs:116]: Shouldn't we create a missile layer?

File: ryot/src/bevy_ryot/drawing/mod.rs [hunk 2]
```diff
@@ -111,6 +135,58 @@ impl DrawingBundle {
...
+/// The moving is removed from the map when the duration is over by default.
+#[derive(Bundle, Debug, Clone)]
+pub struct MovementBundle {
+    pub drawing: DrawingBundle,
+    pub movement: SpriteMovement,
+    pub direction: Directional,
+}
+
+impl MovementBundle {
+    pub fn from_drawing(
```
Comment (lgrossi) [inline ryot/src/bevy_ryot/drawing/mod.rs:151]: from_drawing feels wrong here, maybe new would fit better?

File: ryot/src/bevy_ryot/drawing/mod.rs [hunk 3]
```diff
@@ -100,6 +103,27 @@ impl DrawingBundle {
         )
     }
 
+    pub fn effect(tile_pos: TilePosition, id: u32) -> Self {
```
Comment (lgrossi) [inline ryot/src/bevy_ryot/drawing/mod.rs:106]: Also, to be more cip agnostic, we might want to always receive the layer?

File: ryot/src/directional.rs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+use serde::{Deserialize, Serialize};
+use strum::EnumIter;
+
+#[cfg(feature = "bevy")]
+use bevy::ecs::component::Component;
+
+use crate::position::TilePosition;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy, Default, Debug)]
+#[cfg_attr(feature = "bevy", derive(Serialize, Deserialize))]
```
Comment (lgrossi) [inline ryot/src/directional.rs:13]: We might wanna add a "serde" feature for Ser/De attributes. This is quite standard in Rust.

File: ryot/src/directional.rs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            OrdinalDirection::West => [-1, 0].into(),
+            OrdinalDirection::NorthWest => [-1, 1].into(),
+            OrdinalDirection::None => [0, 0].into(),
+        }
+    }
+}
+
+impl From<IVec2> for OrdinalDirection {
+    fn from(value: IVec2) -> Self {
+        match value.clamp(IVec2::splat(-1), IVec2::splat(1)) {
```
Comment (lgrossi) [inline ryot/src/directional.rs:94]: this is beautiful

File: ryot/src/directional.rs (hunk 4 above)
Comment (luan) [reply ryot/src/directional.rs:13]: We derive these basically everywhere now. I think a separate commit/PR pulling all of them into a feature would make more sense. I pulled these out of "bevy" for now

File: ryot/src/directional.rs (hunk 5 above)
Comment (luan) [reply ryot/src/directional.rs:94]: <3

**Pull request merged**